diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -692,16 +789,20 @@ int DifferentialFunctionComparator::cmpBasicBlocks(
             // ignored (currently allocas and casts). This may help to handle
             // some small changes that do not affect semantics (it is also
             // useful in combination with function inlining).
-            if (mayIgnore(&*InstL) || mayIgnore(&*InstR)) {
+            bool ignoredL = mayIgnore(&*InstL, true);
+            bool ignoredR = ignoredL ? false : mayIgnore(&*InstR, true);
+
+            if (ignoredL || ignoredR) {
                 // Reset serial counters
                 sn_mapL.erase(&*InstL);
                 sn_mapR.erase(&*InstR);
                 // One of the compared operations will be skipped and the
                 // comparison will be repeated.
-                if (mayIgnore(&*InstL))
+                if (ignoredL) {
                     InstL++;
-                else
+                } else {
                     InstR++;
+                }
```

Comment:
Maybe a more readable style would be:

    if (mayIgnore(&*InstL, true)) {
        sn_mapL.erase(&*InstL);
        sn_mapR.erase(&*InstR);
        InstL++;
        continue;
    }
    if (mayIgnore(&*InstR, true)) {
        sn_mapL.erase(&*InstL);
        sn_mapR.erase(&*InstR);
        InstR++;
        continue;
    }

Reply:
Replace `mayIgnore` by `maySkipInstruction` that is proposed above.